### PR TITLE
Add tests for DistributionStrategySLE12SP1 model 

### PR DESCRIPTION
**Pull request by saraycp**

### CODE REVIEW

File: src/api/spec/models/obs_factory/distribution_strategy_sle12_sp1_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  describe '#openqa_version' do
+    it { expect(subject.openqa_version).to eq('SLES 12 SP1') }
+  end
+
+  describe '#openqa_iso' do
+    let(:iso_filename) { 'ISO-SUSE-SLE-12.x86_64-SP1-Build16.64.iso' }
+    before do
+      allow_any_instance_of(ObsFactory::DistributionStrategySLE12SP1).to receive(:project_iso).and_return(iso_filename)
```
Comment (coolo) [inline src/api/spec/models/obs_factory/distribution_strategy_sle12_sp1_spec.rb]: now that looks wrong. Please use some real world example from SP4's A. If you check, the isos are rather named Test-Server-DVD-x86_64-Build19.2-Media.iso and are queried on openqa as SLE12-SP4-Staging:A-Test-Server-DVD-x86_64-BuildA.19.2-Media.iso

File: src/api/spec/models/obs_factory/distribution_strategy_sle12_sp1_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    end
+
+    it { expect(subject.openqa_iso(staging_project)).to eq('SLE12-SP1-Staging:A-Test-Server-DVD-x86_64-BuildA.16.64.iso') }
+
+    context 'without project iso' do
+      let(:iso_filename) { nil }
```
Comment (bgeuken) [inline src/api/spec/models/obs_factory/distribution_strategy_sle12_sp1_spec.rb]: You can drop this. It's not used anywhere

Comment (coolo) [reply src/api/spec/models/obs_factory/distribution_strategy_sle12_sp1_spec.rb]: I was thinking the same, but it's actually very clever as it's used in the shared before hook

File: src/api/spec/models/obs_factory/distribution_strategy_sle12_sp1_spec.rb (hunk 1 above)
Comment (coolo) [reply src/api/spec/models/obs_factory/distribution_strategy_sle12_sp1_spec.rb]: I don't mind, but it's really SP1+ - what I said was true for SP1 as well.

**Pull request merged**